py/desispec/io/spectra.py

```diff
@@ -186,7 +186,25 @@ def write_spectra(outfile, spec, units=None):
     return outfile
 
 
-def read_spectra(infile, single=False):
+def read_spectra(
+    infile,
+    single=False,
+    targetids=None,
+    skip_hdu={
+        "FIBERMAP": False,
+        "EXP_FIBERMAP": False,
+        "SCORES": False,
+        "EXTRA_CATALOG": False,
+        "MASK": False,
+        "RESOLUTION": False,
+    },
+    select_columns={
+        "FIBERMAP": None,
+        "EXP_FIBERMAP": None,
+        "SCORES": None,
+        "EXTRA_CATALOG": None,
+    },
+):
```

Comment:
I like reading all columns by default while providing a flexible way to specify subsets of columns to read per-HDU, but similar to the previous comment let's also allow the user not to specify an HDU at all and have that default to reading all the columns.  This could be implemented by a snippet at the start of the function:
```python
for extname in ('FIBERMAP', 'EXP_FIBERMAP', 'SCORES', 'EXTRA_CATALOG'):
    if extname not in select_columns:
        select_columns[extname] = None
```

Reply:
updated to support extnames missing from user-provided `select_columns` dict.